pkg/codegen/python/gen.go

```diff
@@ -794,6 +783,49 @@ func (mod *modContext) genConfig(variables []*schema.Property) (string, error) {
 	return w.String(), nil
 }
 
+func genConfigFetch(configVar *schema.Property) (string, error) {
+	getFunc := "get"
+	unwrappedType := codegen.UnwrapType(configVar.Type)
+	switch unwrappedType {
+	case schema.BoolType:
+		getFunc = "get_bool"
+	case schema.IntType:
+		getFunc = "get_int"
+	case schema.NumberType:
+		getFunc = "get_float"
+	}
+
+	configFetch := fmt.Sprintf("__config__.%s('%s')", getFunc, configVar.Name)
+	if configVar.DefaultValue != nil {
+		v, err := getDefaultValue(configVar.DefaultValue, unwrappedType)
```

Comment:
So I've looked at `getDefaultValue` func. How do we know that the `DefaultValue` it receives is conformant to `schema.Type`? It does not seem that it currently checks or returns any error in that case. Is this checked somewhere else? Presumably we would want to let the schema author know they're issuing a malformed schema if they specify a default value that does not typecheck against the type they specify.

```
func getDefaultValue(dv *schema.DefaultValue, t schema.Type) (string, error) {
```

Reply:
This is off topic for the PR but I'm still curious. If we don't know I can take a ticket to the side to verify this separately.

Reply:
I'm not sure about this, feel free to dig in 😄